core/scripts/mtx-changer.in

```diff
@@ -131,166 +132,223 @@ wait_for_drive() {
 
-      if test ${offline} -eq 1 ; then
-        mt -f $device offline
-      fi
-      if test ${offline_sleep} -ne 0 ; then
-        sleep ${offline_sleep}
-      fi
-      make_err_file
-      ${MTX} -f $ctl unload $slot $drive 2>${ERRFILE}
-      rtn=$?
-      cat ${ERRFILE}
-      rm -f ${ERRFILE} >/dev/null 2>&1
-      exit $rtn
-      ;;
+    if test ${offline} -eq 1; then
+      mt -f "${device}" offline
+    fi
+    if test ${offline_sleep} -ne 0; then
+      sleep ${offline_sleep}
+    fi
+    make_err_file
+    ${MTX} -f "${ctl}" unload ${slot} ${drive} 2>"${ERRFILE}"
+    rtn=$?
+    cat "${ERRFILE}"
+    rm -f "${ERRFILE}" >/dev/null 2>&1
+    exit $rtn
+    ;;
 
-   load)
-      debug "Doing mtx -f $ctl load $slot $drive"
-      make_err_file
-      ${MTX} -f $ctl load $slot $drive 2>${ERRFILE}
-      rtn=$?
-      if test ${load_sleep} -ne 0 ; then
-        sleep ${load_sleep}
-      fi
-      wait_for_drive $device
-      cat ${ERRFILE}
-      rm -f ${ERRFILE} >/dev/null 2>&1
-      exit $rtn
-      ;;
+  load)
+    debug "Doing mtx -f ${ctl} load ${slot} ${drive}"
+    make_err_file
+    ${MTX} -f "${ctl}" load ${slot} ${drive} 2>"${ERRFILE}"
+    rtn=$?
+    if test ${load_sleep} -ne 0; then
+      sleep ${load_sleep}
+    fi
+    wait_for_drive ${device}
+    cat "${ERRFILE}"
+    rm -f "${ERRFILE}" >/dev/null 2>&1
+    exit $rtn
+    ;;
 
-   list)
-      debug "Doing mtx -f $ctl -- to list volumes"
-      make_temp_file
-      if test ${inventory} -ne 0 ; then
-        ${MTX} -f $ctl inventory
-      fi
-      ${MTX} -f $ctl status >${TMPFILE}
-      rtn=$?
-      if test ${vxa_packetloader} -ne 0 ; then
-        cat ${TMPFILE} | grep " *Storage Element [0-9]*:.*Full" | sed "s/ Storage Element //" | sed "s/Full :VolumeTag=//"
-      else
-	cat ${TMPFILE} | grep " Storage Element [0-9]*:.*Full" | awk "{print \$3 \$4}" | sed "s/Full *\(:VolumeTag=\)*//"
-      fi
-      cat ${TMPFILE} | grep "^Data Transfer Element [0-9]*:Full (Storage Element [0-9]" | awk '{printf "%s:%s\n",$7,$10}'
-      rm -f ${TMPFILE} >/dev/null 2>&1
-      exit $rtn
-      ;;
+  list)
+    debug "Doing mtx -f ${ctl} -- to list volumes"
+    make_temp_file
+    if test ${inventory} -ne 0; then
+      ${MTX} -f "${ctl}" inventory
+    fi
+    ${MTX} -f "${ctl}" status >"${TMPFILE}"
+    rtn=$?
+    #if test ${vxa_packetloader} -ne 0 ; then
+    #  cat "${TMPFILE}" | grep " *Storage Element [0-9]*:.*Full" | sed "s/ Storage Element //" | sed "s/Full :VolumeTag=//"
+    #else
+    #  cat "${TMPFILE}" | grep " Storage Element [0-9]*:.*Full" | awk "{print \$3 \$4}" | sed "s/Full *\(:VolumeTag=\)*//"
+    #fi
+    # We shouldn't have a difference with this pattern
+    awk_code=$(
+cat << '_EOT_'
+    /^ .*Storage Element .*[0-9]:.*Full/ {
+      gsub("[ ].*Storage Element ","",$1)
+      gsub("VolumeTag=","",$3)
+      gsub(/ *$/, "", $3)
+      print $1 ":" $3
+    }
+_EOT_
+    )
+    awk -F ":" "$awk_code" "${TMPFILE}"
```

Comment:
I just realized that this is way more complicated than needed.
We don't need to assign that to a variable, but can just make `awk` read the script from stdin like this:
```suggestion
    awk -f - -F ":" "${TMPFILE}" << '_EOT_'
      /^ .*Storage Element .*[0-9]:.*Full/ {
        gsub("[ ].*Storage Element ","",$1)
        gsub("VolumeTag=","",$3)
        gsub(/ *$/, "", $3)
        print $1 ":" $3
      }
_EOT_
```
That should work for all the other `awk` calls, too.

Reply:
done